api.planx.uk/auth/index.ts

```diff
@@ -37,26 +39,34 @@ const useSendEmailAuth = (
   res: Response,
   next: NextFunction
 ): void => {
-  switch (req.params.template) {
+  const handleInvalidTemplate = (template?: never) => {
+    throw new ServerError({
+      message: "Invalid template",
+      status: 400,
+    });
+  };
+  const template: Template = req.params.template as Template;
+  switch (template) {
     // Requires authorization - can only be triggered by Hasura scheduled events
     case "reminder":
     case "expiry":
+    case "submit":
     case "confirmation":
     case "invite-to-pay":
     case "invite-to-pay-agent":
     case "payment-reminder":
     case "payment-reminder-agent":
     case "payment-expiry":
     case "payment-expiry-agent":
+    case "confirmation-agent":
+    case "confirmation-payee":
       return useHasuraAuth(req, res, next);
     // Public access
     case "save":
+    case "resume":
       return next();
     default: {
-      return next({
-        status: 400,
-        message: "Invalid template",
-      });
+      return handleInvalidTemplate(template);
```

Comment:
I added exhaustive checking here (for parity with a similar function I was introducing) and found missing cases for "submit" and "resume" templates. Are my assumptions about how these should be handled correct?

Reply:
This is a good catch thanks - some of this is handled inconsistently and it would be nice to sort out.

Your assumptions on the permissions for both templates are correct here 👍 

`resume` is left out as there's a `/resume-application` endpoint - I'd suggest we tidy that and update the endpoints so it goes through this middleware and then on to `resumeApplication()`.

`submit` is the same - it has it's own endpoint at `app.post("/email-submission/:localAuthority")`. Untangling that may be a little more involved as we need to calculate the correct local authority for a session but that's something that will be required here also - https://github.com/theopensystemslab/planx-new/pull/1709

Reply:
@DafyddLlyr I've updated how this is handled but I'm not clear on what is needed around "submit". If there is still some remaining work to refactor, could it be a separate PR since it's not directly related to confirmation emails?  I'd be happy to catch-up about this to better understand the requirements.